net/net.go

```diff
@@ -127,48 +131,71 @@ func (t *net) Close() (err error) {
 		}
 	}
 
-	weakClose("DAGService", t.DAGService)
-	weakClose("host", t.host)
-	weakClose("threadstore", t.store)
+	weakClose("DAGService", n.DAGService)
+	weakClose("host", n.host)
+	weakClose("threadstore", n.store)
 
-	t.bus.Discard()
-	t.cancel()
+	n.bus.Discard()
+	n.cancel()
 
 	if len(errs) > 0 {
 		return fmt.Errorf("failed while closing net; err(s): %q", errs)
 	}
 
-	t.pullLock.Lock()
-	defer t.pullLock.Unlock()
+	n.pullLock.Lock()
+	defer n.pullLock.Unlock()
 	// Wait for all thread pulls to finish
-	for _, semaph := range t.pullLocks {
+	for _, semaph := range n.pullLocks {
 		semaph <- struct{}{}
 	}
 	return nil
 }
 
-func (t *net) Host() host.Host {
-	return t.host
+func (n *net) Host() host.Host {
+	return n.host
 }
 
-func (t *net) Store() lstore.Logstore {
-	return t.store
+func (n *net) Store() lstore.Logstore {
+	return n.store
 }
 
-func (t *net) GetHostID(_ context.Context) (peer.ID, error) {
-	return t.host.ID(), nil
+func (n *net) GetHostID(_ context.Context) (peer.ID, error) {
+	return n.host.ID(), nil
 }
 
-func (t *net) CreateThread(_ context.Context, creds thread.Credentials, opts ...core.KeyOption) (info thread.Info, err error) {
-	id := creds.ThreadID()
-	if err = t.ensureUnique(id); err != nil {
+func (n *net) GetToken(_ context.Context, identity thread.Identity) (tok thread.Token, err error) {
```

Comment:
Here's the start of the interesting bits. This method takes an _identity_ interface which allows it to sign a random message payload, or challenge. You might be thinking, but shouldn't the signing happen on the client? It does, because this is just exposing an opaque request for a signature. We can't actually get private key here.